platform/ethereum/compound/compound.go

```diff
@@ -0,0 +1,193 @@
+package compound
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+)
+
+// Compound Lending Provider
+
+type CompoundClient interface {
+	GetAccounts(addresses []string) ([]Account, error)
+	GetCTokensCached(tokenAddresses []string, cacheExpiry time.Duration) (CTokenResponse, error)
+}
+
+type Provider struct {
+	client CompoundClient
+}
+
+const (
+	providerName                   = "compound"
+	cacheValiditySec time.Duration = 15 * time.Second
+)
+
+func Init(api string) *Provider {
+	return &Provider{
+		client: &Client{blockatlas.InitClient(api)},
+	}
+}
+
+func InitForTest(client CompoundClient) *Provider {
+	return &Provider{client: client}
+}
+
+func (p *Provider) Name() string {
+	return providerName
+}
+
+// GetProviderInfo return static info about the lending provider, such as name and asset classes supported.
+func (p *Provider) GetProviderInfo() (blockatlas.LendingProvider, error) {
+	provider := blockatlas.LendingProvider{
+		ID: "compound",
+		Info: blockatlas.StakeValidatorInfo{
+			Name:        providerName,
+			Description: "Compound Decentralized Finance Protocol",
+			Image:       "https://compound.finance/images/compound-logo.svg",
+			Website:     "https://compound.finance",
+		},
+		Type:   blockatlas.ProviderTypeLending,
+		Assets: make([]blockatlas.AssetInfo, 0),
+	}
+	assets, err := p.getAssetInfos(false)
+	if err == nil {
+		provider.Assets = assets
+	}
+	return provider, nil
+}
+
+func (p *Provider) GetAssets() ([]blockatlas.AssetInfo, error) {
+	return p.getAssetInfos(true)
+}
+
+// GetAccountLendingContracts return current contract details for a given address.
+// req.Assets: List asset IDs to consider, or empty for all
+func (p *Provider) GetAccountLendingContracts(req blockatlas.AccountRequest) ([]blockatlas.AccountLendingContracts, error) {
+	accounts, err := p.client.GetAccounts(req.Addresses)
+	if err != nil {
+		return nil, err
+	}
+	ret := make([]blockatlas.AccountLendingContracts, 0)
```

Comment:
ret?

Reply:
Renamed to contracts and accountContracts.